dirsrvtests/tests/suites/clu/dbscan_test.py

```diff
@@ -0,0 +1,253 @@
+from difflib import context_diff
+
+pytestmark = pytest.mark.tier0
+
+logging.getLogger(__name__).setLevel(logging.DEBUG)
+log = logging.getLogger(__name__)
+
+DEBUGGING = os.getenv("DEBUGGING", default=False)
+
+
+class CalledProcessUnexpectedReturnCode(subprocess.CalledProcessError):
+    def __init__(self, result, expected_rc):
+        super().__init__(cmd=result.args, returncode=result.returncode, output=result.stdout, stderr=result.stderr)
+        self.expected_rc = expected_rc
+        self.result = result
+
+    def __str__(self):
+        return f'Command {self.result.args} returned {self.result.returncode} instead of {self.expected_rc}'
+
+
+class DbscanPaths:
+    @staticmethod
+    def list_instances(inst, dblib, dbhome):
+        # compute db instance pathnames
+        instances = dbscan(['-D', dblib, '-L', dbhome], inst=inst).stdout
+        dbis = []
+        if dblib == 'bdb':
+            pattern = r'^ (.*) $'
+            prefix = f'{dbhome}/'
+        else:
+            pattern = r'^ (.*) flags:'
+            prefix = f''
+        for match in re.finditer(pattern, instances, flags=re.MULTILINE):
+            dbis.append(prefix+match.group(1))
+        return dbis
+
+    @staticmethod
+    def list_options(inst):
+        # compute supported options
+        options = []
+        usage = dbscan(['-h'], inst=inst, expected_rc=None).stdout
+        pattern = r'^\s+(?:(-[^-,]+), +)?(--[^ ]+).*$'
+        for match in re.finditer(pattern, usage, flags=re.MULTILINE):
+            for idx in range(1,3):
+                if not match.group(idx) is None:
+                    options.append(match.group(idx))
+        return options
+
+    def __init__(self, inst):
+        dblib = inst.get_db_lib()
+        dbhome = inst.ds_paths.db_home_dir
+        self.inst = inst
+        self.dblib = dblib
+        self.dbhome = dbhome
+        self.options = DbscanPaths.list_options(inst)
+        self.dbis = DbscanPaths.list_instances(inst, dblib, dbhome)
+        self.ldif_dir = inst.ds_paths.ldif_dir
+
+    def get_dbi(self, attr, backend='userroot'):
+        for dbi in self.dbis:
+            if f'{backend}/{attr}.'.lower() in dbi.lower():
+                return dbi
+        raise KeyError(f'Unknown dbi {backend}/{attr}')
+
+    def __repr__(self):
+        attrs = ['inst', 'dblib', 'dbhome', 'ldif_dir', 'options', 'dbis' ]
+        res = ", ".join(map(lambda x: f'{x}={self.__dict__[x]}', attrs))
+        return f'DbscanPaths({res})'
+
+
+def dbscan(args, inst=None, expected_rc=0):
+    if inst is None:
+        prefix = os.environ.get('PREFIX', "")
+        prog = f'{prefix}/bin/dbscan'
+    else:
+        prog = os.path.join(inst.ds_paths.bin_dir, DBSCAN)
+    args.insert(0, prog)
+    output = subprocess.run(args, encoding='utf-8', stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    log.debug(f'{args} result is {output.returncode} output is {output.stdout}')
+    if expected_rc != output.returncode and not expected_rc is None:
```

Comment:
```suggestion
    if expected_rc != output.returncode and expected_rc is not None:
```

Reply:
This is a minor nitpick, but I think `is not None` is better readable. `not expected_rc` is also viable, so `not expected_rc is None:` can be misread accidentally by humans. In this case, `is not None` is straightforward.

Feel free to leave it as it is if it's your style. It compiles to the same code anyway.

Reply:
Also, probably, `expected_rc is not None` should be checked first
```suggestion
    if expected_rc is not None and expected_rc != output.returncode:
```